### PR TITLE
ci: move to semantic-release for versioning refs: SDKUI-11

### PR DESCRIPTION
**Pull request by CerDennis**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+ssh://git@github.com:Zextras/carbonio-ui-sdk.git"
+    "url": "git+https://github.com/zextras/carbonio-ui-sdk.git"
   },
   "engines": {
     "node": "v18",
```
Comment (beawar) [inline package.json]: We should add the compatibility with node 20 too

File: Jenkinsfile [hunk 2]
```diff
@@ -234,24 +70,10 @@ pipeline {
...
-                   echo "isDevBuild: ${isDevBuild}"
-                   pkgName = getPackageName()
-                   echo "pkgName: ${pkgName}"
-                   pkgDescription = getPackageDescription()
-                   echo "pkgDescription: ${pkgDescription}"
-                   pkgFullVersion = getPackageVersion()
-                   echo "pkgFullVersion: ${pkgFullVersion}"
                 }
                 withCredentials([
                     usernamePassword(
```
Comment (beawar) [inline Jenkinsfile]: The login and stash blocks can be removed too

File: package.json [hunk 3]
```diff
@@ -12,10 +12,10 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+ssh://git@github.com:Zextras/carbonio-ui-sdk.git"
+    "url": "git+https://github.com/zextras/carbonio-ui-sdk.git"
   },
   "engines": {
-    "node": "v18",
+    "node": "v20",
```
Comment (beawar) [inline package.json]: By replacing it, we are removing support for node 18. We could keep both node 18 and node 20

File: package.json [hunk 4]
```diff
@@ -12,10 +12,10 @@
   },
   "repository": {
     "type": "git",
-    "url": "git+ssh://git@github.com:Zextras/carbonio-ui-sdk.git"
+    "url": "git+https://github.com/zextras/carbonio-ui-sdk.git"
   },
   "engines": {
-    "node": "v18",
+    "node": ">=18.0.0 <19.0.0 || >=20.0.0 <21.0.0",
```
Comment (beawar) [inline package.json:18]: `^18 || ^20` is an alternative

**Review by beawar — APPROVED** [review]

**Review by lisaparma — APPROVED** [review]

**Review by nubsthead — APPROVED** [review]

**Pull request merged**